client/my-sites/site-settings/controller.js

```diff
@@ -124,6 +125,16 @@ export function legacyRedirects( context, next ) {
 	next();
 }
 
+export function redirectToGeneral( context, next ) {
+	const site = getSiteFragment( context.path );
+
+	if ( site ) {
+		return page.redirect( `/settings/general/${ site }` );
```

Comment:
There's already a similar route defined for site-less settings: 49a74f8f4497bacf9de3fdfe9860e44f62bfa366f8cc4e6R28

Should we combine the logic for those 2?